crates/db_views/community/src/multi_community.rs

```diff
@@ -0,0 +1,151 @@
+use crate::{MultiCommunityView, MultiCommunityViewApub};
+use diesel::{
+  dsl::{not, sql},
+  result::Error,
+  sql_types::{Array, Integer, Text},
+  ExpressionMethods,
+  NullableExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::RunQueryDsl;
+use lemmy_db_schema::{
+  newtypes::{CommunityId, DbUrl, MultiCommunityId},
+  utils::{get_conn, DbPool},
+};
+use lemmy_db_schema_file::schema::{community, multi_community, multi_community_entry, person};
+
+pub enum ReadParams {
+  Id(MultiCommunityId),
+  ApId(DbUrl),
+}
+
+impl MultiCommunityView {
+  pub async fn read(
+    pool: &mut DbPool<'_>,
+    params: ReadParams,
+  ) -> Result<MultiCommunityView, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let mut query = multi_community::table
+      .left_join(person::table)
+      .left_join(multi_community_entry::table.inner_join(community::table))
+      .group_by(multi_community::id)
+      .filter(not(community::removed))
+      .filter(not(community::deleted))
```

Comment:
Somehow this is not working and always returns not_found. I already tried adding `.nullable()` but same. Any other ideas?

Reply:
Use filter(community::removed.or(community::deleted).is_distinct_from(true)) so it works even if community is null.

Reply:
Its because you're trying to take a single multi_community, and join it to a list of things, which isn't how SQL works.

IMO you don't need this view at all. In the `multi_community/get` route, fetch the multi_community, and the list of `communities` separately. There's no need to join the multi_community to each community fetched. IE `MultiCommunity::list_communities(multi_community_id)`